xray/services/scan.go

```diff
@@ -192,12 +193,17 @@ type GraphNode struct {
 }
 
 // FlattenGraph creates a map of dependencies from the given graph, and returns a flat graph of dependencies with one level.
-func FlattenGraph(graph []*GraphNode) []*GraphNode {
+func FlattenGraph(graph []*GraphNode) ([]*GraphNode, error) {
 	allDependencies := map[string]*GraphNode{}
 	for _, node := range graph {
 		populateUniqueDependencies(node, allDependencies)
 	}
-	return []*GraphNode{{Id: "root", Nodes: maps.Values(allDependencies)}}
+	jsonList, err := json.Marshal(maps.Keys(allDependencies))
+	if err != nil {
+		return nil, errorutils.CheckError(err)
+	}
+	log.Debug("Flat dependencies list:\n" + clientutils.IndentJsonArray(jsonList))
```

Comment:
Flat dependencies list: --> Flattened list dependencies to be scanned by Xray:

I also suggest we run the following lines of code only if the log is set to debug, to save ptocessing time. For this, may need to add a new `log.IsDebugEnabled` API - 
```
	jsonList, err := json.Marshal(maps.Keys(allDependencies))
	if err != nil {
		return nil, errorutils.CheckError(err)
	}
```